### DISCUSSION TITLE
ERROR: Material property is not defined on block 0

### DISCUSSION THREAD

**Discussion by LiuPengPeter** [idaholab/moose#16079] [Q&A General] [answered]
Hello guys,
*** ERROR ***
Material property 'v_Coefficient', requested by 'StudyCoefficientGradientV' is not defined on block 0
[code.zip](https://github.com/idaholab/moose/files/5491219/code.zip)

I checked my exercise codes for many times  and still couldn't solve the problem... Could anyone help?

**Comment by WilkAndy** [comment]
This is probably not due to your C code.  It is probably due to your input file.  Your input file needs a Material block that will produce a MaterialProperty called "v_Coefficient", since your "StudyCoefficientGradientV" Material needs that MaterialProperty.

**Reply by LiuPengPeter** [reply]
My .i file is also in the zip.

`[Kernels]
  [./DiffusionU]
  type = Diffusion
  variable = u
  [../]
  [./StudyCoefficientGradientV]
  type = StudyCoefficientGradientV
  variable = u
  **D = v_Coefficient**
  [../]

`[Materials]
  [./StudyMaterialCoefficientGradientV]
  type = StudyMaterialCoefficientGradientV
  **D = v_Coefficient**
  v = v
  [../]

Is there something wrong?

**Comment by cpgr** [comment, marked as answer]
The problem is in your `StudyMaterialCoefficientGradientV` material. Take a look at this bit:
```
   _v_coefficient_name(getParam<MaterialPropertyName>("D")),
   _v_coefficient(declareProperty<RealGradient>("_v_coefficient_name"))
```

You are declaring a property called `_v_coefficient_name` (that actual string), not whatever you enter in the parameter `D` in your input file. Simply remove the `""` in the second line and it should work as you are now declaring a property with the name given in the input file.

**Reply by LiuPengPeter** [reply]
Thanks a lot!

**Reply by WilkAndy** [reply]
@cpgr saves the day again!
